python/friesian/src/bigdl/friesian/feature/utils.py

```diff
@@ -214,6 +214,66 @@ def str_to_list(arg, arg_name):
     return arg
 
 
+def featuretable_to_xshards(tbl, convert_cols=None):
+    from bigdl.friesian.feature import FeatureTable
+    from bigdl.orca.data import SparkXShards
+    from pyspark.ml.linalg import DenseVector
+    import numpy as np
+    assert isinstance(tbl, FeatureTable), "This function only supports friesian FeatureTable"
+    # TODO: partition < node num
+    if convert_cols is None:
+        convert_cols = tbl.columns
+    if convert_cols and not isinstance(convert_cols, list):
+        convert_cols = [convert_cols]
+
+    schema = tbl.df.schema
+    np_type_dict = dict()
+    for col in convert_cols:
+        assert col in tbl.columns, "Column name " + col + " does not exist in the FeatureTable"
+        feature_type = schema[col].dataType
+        np_type = df_type_to_np_type(feature_type)
+        if np_type:
+            np_type_dict[col] = np_type
+
+    def to_numpy_dict(iter):
+        rows = list(iter)
```

Comment:
this can take a lot of memory if we have a very large partition. 

Maybe try to break the partition into smaller sizes if `OrcaContext._shard_size` is set.
https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/common.py#L106